app/models/katello/upstream_pool.rb

```diff
@@ -1,12 +1,20 @@
-require 'katello/resources/candlepin'
-
 module Katello
   class UpstreamPool < OpenStruct
     CP_POOL = Resources::Candlepin::UpstreamPool
 
     class << self
       def fetch_pools(params)
-        pools = JSON.parse(CP_POOL.get(params: cp_request_params.deep_merge(params)))
+        response = CP_POOL.get(params: cp_request_params.deep_merge(params))
+
+        pools = response_to_pools(response)
+        {
+          pools: pools,
+          total: response.headers[total_count_header] || pools.count
+        }
```

Comment:
I'd prefer this in a method, but it could be refactored later when we have other places we want to use it. Honestly, I don't know if this `x_total_count` response is (or will be) typical of Candlepin responses, so I don't know if we'll need reusable code. :)